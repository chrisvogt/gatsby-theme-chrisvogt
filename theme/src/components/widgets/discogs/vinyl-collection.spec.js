import React from 'react'
import renderer from 'react-test-renderer'

import VinylCollection from './vinyl-collection'

const mockReleases = [
  {
    id: 28461454,
    basicInformation: {
      id: 28461454,
      title: 'The Rise & Fall Of A Midwest Princess',
      year: 2023,
      artists: [{ name: 'Chappell Roan' }],
      cdnThumbUrl: 'https://example.com/thumb1.jpg',
      resourceUrl: 'https://discogs.com/release/123'
    }
  },
  {
    id: 33129744,
    basicInformation: {
      id: 33129744,
      title: 'Brat And It\'s Completely Different',
      year: 2025,
      artists: [{ name: 'Charli XCX' }],
      cdnThumbUrl: 'https://example.com/thumb2.jpg',
      resourceUrl: 'https://discogs.com/release/456'
    }
  }
]

describe('VinylCollection', () => {
  it('renders loading state', () => {
    const tree = renderer
      .create(<VinylCollection isLoading={true} releases={[]} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with vinyl releases', () => {
    const tree = renderer
      .create(<VinylCollection isLoading={false} releases={mockReleases} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders empty state when no releases', () => {
    const tree = renderer
      .create(<VinylCollection isLoading={false} releases={[]} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
}) 