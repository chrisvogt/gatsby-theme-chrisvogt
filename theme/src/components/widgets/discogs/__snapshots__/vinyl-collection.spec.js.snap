// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`VinylCollection renders empty state when no releases 1`] = `
<div
  className="css-16ilj6a"
>
  <div
    className="css-ccroti"
  >
    <h3
      className="css-1pgqsji"
    >
      Vinyl Collection
    </h3>
  </div>
  <p
    className="css-1avyp1d"
  >
    My owned vinyl records from Discogs.
  </p>
  <div
    className="css-1alnzao"
  >
    <div
      className="css-ngfim1"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    />
  </div>
</div>
`;

exports[`VinylCollection renders loading state 1`] = `
<div
  className="css-16ilj6a"
>
  <div
    className="css-ccroti"
  >
    <h3
      className="css-1pgqsji"
    >
      Vinyl Collection
    </h3>
  </div>
  <p
    className="css-1avyp1d"
  >
    My owned vinyl records from Discogs.
  </p>
  <div
    className="css-1alnzao"
  >
    <div
      className="css-ngfim1"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    />
  </div>
</div>
`;

exports[`VinylCollection renders with vinyl releases 1`] = `
<div
  className="css-16ilj6a"
>
  <div
    className="css-ccroti"
  >
    <h3
      className="css-1pgqsji"
    >
      Vinyl Collection
    </h3>
  </div>
  <p
    className="css-1avyp1d"
  >
    My owned vinyl records from Discogs.
  </p>
  <div
    className="css-1alnzao"
  >
    <div
      className="css-itz9zd"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="css-gjogkf"
      >
        <div
          className="vinyl-collection_grid null css-4004kc"
        >
          <div
            className="css-139dwiw"
          >
            <div
              className="vinyl-record css-4gxhk2"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              title="The Rise & Fall Of A Midwest Princess (2023) - Chappell Roan"
            >
              <div
                className="css-wioo5x"
              >
                <div
                  className="vinyl-record_image css-18n965x"
                >
                  <img
                    alt="The Rise & Fall Of A Midwest Princess album cover"
                    className="css-1ybonlr"
                    crossOrigin="anonymous"
                    loading="lazy"
                    src="https://example.com/thumb1.jpg"
                  />
                </div>
                <div
                  className="vinyl-record_caption css-11wixuo"
                >
                  <span
                    className="css-kazeaq"
                  >
                    The Rise & Fall Of A Midwest Princess
                  </span>
                  <span
                    className="css-53gxr7"
                  >
                    Chappell Roan
                  </span>
                  <span
                    className="css-yztu9z"
                  >
                    2023
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="css-139dwiw"
          >
            <div
              className="vinyl-record css-4gxhk2"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              title="Brat And It's Completely Different (2025) - Charli XCX"
            >
              <div
                className="css-wioo5x"
              >
                <div
                  className="vinyl-record_image css-18n965x"
                >
                  <img
                    alt="Brat And It's Completely Different album cover"
                    className="css-1ybonlr"
                    crossOrigin="anonymous"
                    loading="lazy"
                    src="https://example.com/thumb2.jpg"
                  />
                </div>
                <div
                  className="vinyl-record_caption css-11wixuo"
                >
                  <span
                    className="css-kazeaq"
                  >
                    Brat And It's Completely Different
                  </span>
                  <span
                    className="css-53gxr7"
                  >
                    Charli XCX
                  </span>
                  <span
                    className="css-yztu9z"
                  >
                    2025
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
